code/game/gamemodes/cult/runes.dm

```diff
@@ -1,1046 +1,1003 @@
+	icon_state = "2"
+	req_keyword = 1
+	var/listkey
+
+/obj/effect/rune/teleport/New(loc, set_keyword)
+	..()
+	var/area/A = get_area(src)
+	var/locname = initial(A.name)
+	listkey = set_keyword ? "[set_keyword] [locname]":"[locname]"
+	teleport_runes += src
+
+/obj/effect/rune/teleport/Destroy()
+	teleport_runes -= src
+	return ..()
+
+/obj/effect/rune/teleport/invoke(var/list/invokers)
+	var/mob/living/user = invokers[1] //the first invoker is always the user
+	var/list/potential_runes = list()
+	var/list/teleportnames = list()
+	var/list/duplicaterunecount = list()
+	for(var/R in teleport_runes)
+		var/obj/effect/rune/teleport/T = R
+		var/resultkey = T.listkey
+		if(resultkey in teleportnames)
+			duplicaterunecount[resultkey]++
+			resultkey = "[resultkey] ([duplicaterunecount[resultkey]])"
 		else
-			to_chat(usr, "\red The sacrifical corpse is not dead. You must free it from this world of illusions before it may be used.")
-		return fizzle()
-
-	var/mob/dead/observer/ghost
-	for(var/mob/dead/observer/O in loc)
-		if(!O.client)	continue
-		if(O.mind && O.mind.current && O.mind.current.stat != DEAD)	continue
-		ghost = O
-		break
-
-	if(!ghost)
-		to_chat(usr, "\red You require a restless spirit which clings to this world. Beckon their prescence with the sacred chants of Nar-Sie.")
-		return fizzle()
-
-	corpse_to_raise.revive()
-
-	corpse_to_raise.key = ghost.key	//the corpse will keep its old mind! but a new player takes ownership of it (they are essentially possessed)
-									//This means, should that player leave the body, the original may re-enter
-	usr.say("Pasnar val'keriam usinar. Savrae ines amutan. Yam'toth remium il'tarat!")
-	corpse_to_raise.visible_message("\red [corpse_to_raise]'s eyes glow with a faint red as he stands up, slowly starting to breathe again.", \
-	"\red Life... I'm alive again...", \
-	"\red You hear a faint, slightly familiar whisper.")
-	body_to_sacrifice.visible_message("\red [body_to_sacrifice] is torn apart, a black smoke swiftly dissipating from his remains!", \
-	"\red You feel as your blood boils, tearing you apart.", \
-	"\red You hear a thousand voices, all crying in pain.")
-	body_to_sacrifice.gib()
-
-//			if(ticker.mode.name == "cult")
-//				ticker.mode:add_cultist(corpse_to_raise.mind)
-//			else
-//				ticker.mode.cult |= corpse_to_raise.mind
-
-	to_chat(corpse_to_raise, "<font color=\"purple\"><b><i>Your blood pulses. Your head throbs. The world goes red. All at once you are aware of a horrible, horrible truth. The veil of reality has been ripped away and in the festering wound left behind something sinister takes root.</b></i></font>")
-	to_chat(corpse_to_raise, "<font color=\"purple\"><b><i>Assist your new compatriots in their dark dealings. Their goal is yours, and yours is theirs. You serve the Dark One above all else. Bring It back.</b></i></font>")
-	return
+			teleportnames.Add(resultkey)
+			duplicaterunecount[resultkey] = 1
+		if(T != src && (!is_level_reachable(T.z)))
```

Comment:
This conditional should be flipped - the station is reachable, centcomm and the away mission are not